src/provider/cloud-element-templates/ReplaceBehavior.js

```diff
@@ -33,7 +33,15 @@ export default class ReplaceBehavior extends CommandInterceptor {
         return;
       }
 
-      const { appliesTo } = elementTemplate;
+      const { appliesTo, elementType } = elementTemplate;
+
+      if (elementType) {
+        if (!is(newShape, elementType.value)) {
```

Comment:
I wonder what happens when `elementType.value` is undefined. But that's anyway not possible since we are validating templates beforehand.

Reply:
Related: https://github.com/bpmn-io/bpmn-js-properties-panel/pull/648#discussion_r839540764